plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergSplitSource.java

```diff
@@ -311,7 +344,63 @@ private boolean noDataColumnsProjected(FileScanTask fileScanTask)
+    }
+
+    private boolean satisfyAggregations(FileScanTask wholeFileTask)
+    {
+        checkState(fileAggregateValues.isEmpty(), "fileAggregateValues already set");
+
+        // Invoked only for files where unconditionallyProjectedBaseColumns (groupBy) is only partitioning columns
+        ImmutableList.Builder<Block> values = ImmutableList.builder();
+        for (Aggregation aggregation : aggregations) {
+            Optional<Object> value = satisfy(wholeFileTask.file(), aggregation);
+            if (value.isEmpty()) {
+                return false;
+            }
+            values.add(Utils.nativeValueToBlock(aggregation.outputType(), value.get()));
+        }
+        fileAggregateValues = Optional.of(values.build());
+        return true;
+    }
+
+    private Optional<Object> satisfy(DataFile dataFile, Aggregation aggregation)
+    {
+        AggregationType aggregationType = aggregation.aggregationType();
+        return switch (aggregationType) {
+            case COUNT_ALL -> Optional.of(dataFile.recordCount());
+            case COUNT_NON_NULL -> {
+                IcebergColumnHandle argument = getOnlyElement(aggregation.arguments());
+                int fieldId = argument.getId();
+                Long valueCount = firstNonNull(dataFile.valueCounts(), ImmutableMap.<Integer, Long>of()).get(fieldId);
+                Long nullCount = firstNonNull(dataFile.nullValueCounts(), ImmutableMap.<Integer, Long>of()).get(fieldId);
+                if (valueCount == null || nullCount == null) {
+                    yield Optional.empty();
+                }
+                yield Optional.of(valueCount - nullCount);
+            }
+            case MIN, MAX -> {
+                IcebergColumnHandle argument = getOnlyElement(aggregation.arguments());
+                io.trino.spi.type.Type argumentTrinoType = argument.getType();
+                int fieldId = argument.getId();
+                Map<Integer, ByteBuffer> bounds = firstNonNull((aggregationType == MIN) ? dataFile.lowerBounds() : dataFile.upperBounds(), ImmutableMap.of());
+                if (!bounds.containsKey(fieldId)) {
+                    yield Optional.empty();
+                }
+                Type icebergType = fieldIdToType.get(fieldId);
+                Object value = convertIcebergValueToTrino(icebergType, fromByteBuffer(icebergType, bounds.get(fieldId)));
+                if (argumentTrinoType instanceof VarcharType) {
+                    // TODO consult org.apache.iceberg.MetricsModes.MetricsMode
+                    // TODO this is incorrect for max: truncate("abc<max-code-point><max-code-point>...<max-code-point>") -> "abd", so upper bound value being "abd"
+                    //  does not imply the max value is "abd". For this to work Iceberg would need to track upper bound exactness, like Parquet https://github.com/apache/parquet-format/pull/216
+                    if (countCodePoints((Slice) value) >= MIN_MAX_TRUSTED_VARCHAR_LENGTH_LIMIT) {
```

Comment:
This is more aggressive than it needs to be.
Instead of pre-computing the answer to `max(col)` query (which may be correct or rounded), this should be converted into split pruning.
To satisfy max query, we need to open only the splits that can contribute to the max. And within the file, open only those row groups that can contribute to the max.

This would also work for date/time values, which are pretty much always rounded.